recipes/NWM-1km/recipe.py

```diff
@@ -0,0 +1,29 @@
+import pandas as pd
+
+from pangeo_forge_recipes.patterns import ConcatDim, FilePattern
+from pangeo_forge_recipes.recipes import XarrayZarrRecipe
+
+dates = pd.date_range('1979-02-01 03:00', '2020-12-31 21:00', freq='3H')
+
+time_concat_dim = ConcatDim('time', dates, nitems_per_file=1)
+
+URL_FORMAT = (
+    's3://noaa-nwm-retrospective-2-1-pds/model_output/'
+    '{time:%Y}/{time:%Y%m%d%H}00.LDASOUT_DOMAIN1.comp'
+)
+
+
+def make_url(time):
+    return URL_FORMAT.format(time=time)
```

Comment:
@rsignell-usgs, the latest error is:

```
File "/tmp/tmp534qhcdq/recipes/NWM-1km/recipe.py", line 17, in make_url
NameError: name 'URL_FORMAT' is not defined [while running 'Start|cache_input|Reshuffle_000|prepare_target|Reshuffle_001|store_chunk|Reshuffle_002|finalize_target|Reshuffle_003/cache_input/Execute-ptransform-56']
```

which is a version of the same issue almost every PR this week has had 😅 : the unexpected serialization quirk on Google Cloud Dataflow (where we run these jobs), wherein module-level variables are not available within function scopes. The workaround is to define any variable used within a function inside the function body itself. The proposed fix below should resolve this.

```suggestion
def make_url(time):
    URL_FORMAT = (
        's3://noaa-nwm-retrospective-2-1-pds/model_output/'
        '{time:%Y}/{time:%Y%m%d%H}00.LDASOUT_DOMAIN1.comp'
    )
    return URL_FORMAT.format(time=time)
```